### PR TITLE
EDUCATOR-5207: Look up team submissions / workflows by learner rather than by team

### PR DESCRIPTION
**Pull request by jansenk**
**TL;DR -** When a user loads an ora, or when a staff member looks at a student's submission, look up the submission by the student_item rather than by team.

**What changed?**

- In several places we looked up submissions and workflows by a user's current team.
- This has been changed to now look them up by the specific student, that way we can support users who have submitted a submission with a different team than they are currently on.

**What changed?: 2**

- Bumped submission version to grab changes from https://github.com/edx/edx-submissions/pull/127
- added method `add_team_submission_context` to `team_mixin` to unify logic for adding team information (team name and usernames) from a team submission, which is used in `staff_area`
- Changed `message_mixin` to not display error message about needing to be on a team if you already have a submission (and also changed the message to the standard "complete" message if the workflow is in the 'teams' state)
- Changed team_workflow_mixin to look up team_submission_uuid by student_item rather than by team
- some methods in FIleUploadManager now take a team_id parameter, rather than just grabbing team_id from `self.block.team.team_id`, since the team_id we want to grab teams for is potentially not the team the user is currently on
- Lots of test changes

**Developer Checklist**
- [x] Reviewed the [release process](./release_process.md)
- [x] Translations up to date
- [x] JS minified, SASS compiled
- [x] Test suites passing on Jenkins
- [x] Bumped version number in [setup.py](../setup.py) and [package.json](../package.json)

JIRA: [EDUCATOR-5207](https://openedx.atlassian.net/browse/EDUCATOR-5207)

FIY: @edx/masters-devs-gta

### CODE REVIEW

**Comment by jansenk** [comment]
I've got a failing test I've gotta fix and some quality errors, but this is ready for review.

File: openassessment/xblock/team_mixin.py [hunk 1]
```diff
@@ -99,6 +104,41 @@ def valid_access_to_team_assessment(self):
         """
         return self.is_course_staff or self.in_studio_preview or self.has_team()
 
+    def add_team_submission_context(
```
Comment (Andytr1) [inline openassessment/xblock/team_mixin.py:107]: I thought 'get_team_info()' below added that information to context?

File: openassessment/xblock/team_mixin.py [hunk 2]
```diff
@@ -99,6 +104,41 @@ def valid_access_to_team_assessment(self):
...
+            - individual_submission_uuid (string): [optional] the uuid of an individual submission that's a part of
+                                                   the team submission for which we want to add context info
+            - transform_usernames (bool): [optional default: False] If False, context['team_usernames'] will be a list
+                                          of username strings. If True, it will be a string, in the form
+                                          "Username1, Username2, ... UsernameN, and UsernameN+1"
+
+        One of team_submission_uuid and individual_submission_uuid are required, and if they are both provided,
+        individual_submission_uuid will be ignored.
+        """
+        if not any((team_submission_uuid, individual_submission_uuid)):
```
Comment (Andytr1) [inline openassessment/xblock/team_mixin.py:126]: Would it make sense to have this method return team name for anonymous user (see staff_area_mixin.py::353) ?

File: openassessment/xblock/team_mixin.py (hunk 1 above)
Comment (jansenk) [reply openassessment/xblock/team_mixin.py:107]: get_team_info adds information to the context based on team memberships, this specifically is adding that info from a submission

Comment (jansenk) [reply openassessment/xblock/team_mixin.py:107]: but uh, wait, yeah, maybe it should? If there's no submission we call get_team_info which adds additional info to the context ....

Comment (jansenk) [reply openassessment/xblock/team_mixin.py:107]: okay ignore these comments I'm gonna think abut this a bit more

File: openassessment/xblock/test/test_staff_area.py [hunk 3]
```diff
@@ -1291,6 +1339,31 @@ def _setup_xblock_and_create_submission(self, xblock, anonymous_user_id='Bob', t
...
+            student_item = STUDENT_ITEM.copy()
+            student_item["item_id"] = xblock.location
+            return self._create_submission(student_item, {
+                'text': "Text Answer",
+                'file_keys': kwargs.get('file_keys', []),
+                'files_descriptions': kwargs.get('files_descriptions', []),
+                'files_names': kwargs.get('files_names', [])
+            }, ['staff'])
+
+    def _setup_xblock(self, xblock, anonymous_user_id='Bob', team=False, has_team=True):
```
Comment (muselesscreator) [inline openassessment/xblock/test/test_staff_area.py:1363]: I feel like this logic, or at least a good chunk of it, should maybe live in a share-able location for other modules, as setting up the teams service appropriately is v time and code-consuming.

**Comment by muselesscreator** [comment]
Just tested.  After standup conversation about moving the task of re-usability to a later ticket, LGTM.  :shipit:

File: openassessment/xblock/team_mixin.py (hunk 2 above)
Comment (jansenk) [reply openassessment/xblock/team_mixin.py:126]: I'm gonna not do this, in order to keep the two things logically separate. So for now,

get_team_info() gets you the info for the team the user is currently on, 

add_team_submission_context gets you team info specifically from a submission

File: openassessment/xblock/submission_mixin.py [hunk 4]
```diff
@@ -738,9 +740,21 @@ def submission_path_and_context(self):
 
         file_urls = None
 
+        team_id_for_shared_files = None
```
Comment (nsprenkle) [inline openassessment/xblock/submission_mixin.py:743]: Maybe it has to wait until we consolidate the stories in flight but I'd argue for a different naming. This suggests to me that we have different team IDs for files and other parts of the page when I thought we wanted to source most of the page context differently based on whether the team has submitted or not. 
```suggestion
        submission_team_id = None
```

File: openassessment/xblock/test/test_staff_area.py [hunk 5]
```diff
@@ -851,32 +858,13 @@ def test_staff_assessment_form(self, xblock):
...
-        # mock list format
-        def mock_formatter(entries):
-            return ', '.join(entries)
-        formatter.side_effect = mock_formatter
-
-        # Simulate that we are course staff
-        xblock.xmodule_runtime = self._create_mock_runtime(
-            xblock.scope_ids.usage_id, True, False, "Bob"
-        )
+    def test_staff_form_for_team_assessment(self, xblock, ):
```
Comment (nsprenkle) [inline openassessment/xblock/test/test_staff_area.py:862]: Nit: trailing comma
```suggestion
    def test_staff_form_for_team_assessment(self, xblock):
```

File: openassessment/xblock/test/test_staff_area.py [hunk 6]
```diff
@@ -1332,29 +1405,27 @@ def _setup_xblock_and_create_submission(self, xblock, anonymous_user_id='Bob', t
...
-            )
-        else:
-            return self._create_submission(student_item, {
-                'text': "Text Answer",
-                'file_keys': kwargs.get('file_keys', []),
-                'files_descriptions': kwargs.get('files_descriptions', []),
-                'files_names': kwargs.get('files_names', [])
-            }, ['staff'])
+
+            # For both functions, map values in MOCK_TEAM_MEMBER_STUDENT_IDS to values in MOCK_TEAM_MEMBER_USERNAMES,
```
Comment (nsprenkle) [inline openassessment/xblock/test/test_staff_area.py:1414]: This makes me happy 😄

File: openassessment/xblock/team_mixin.py [hunk 7]
```diff
@@ -99,6 +104,41 @@ def valid_access_to_team_assessment(self):
...
+        elif individual_submission_uuid:
+            team_submission = get_team_submission_from_individual_submission(individual_submission_uuid)
+
+        team = self.teams_service.get_team_by_team_id(team_submission['team_id'])
+        context['team_name'] = team.name
+
+        student_ids = get_team_submission_student_ids(team_submission['team_submission_uuid'])
+        usernames = [self.get_username(student_id) for student_id in student_ids]
+        if transform_usernames:
+            usernames = list_to_conversational_format(usernames)
```
Comment (nsprenkle) [inline openassessment/xblock/team_mixin.py:139]: Unrelated rant: I still think `list_to_conversational_format` should be handled in the fronted or, minimally, a template filter. Maybe it shouldn't but it irks me to build display data in the backend.

Comment (muselesscreator) [reply openassessment/xblock/team_mixin.py:139]: When we get react, it won't be.   Problem is the templates

**Comment by muselesscreator** [comment]
Tested file uploads on team change and they work for me :-)

File: openassessment/xblock/team_mixin.py (hunk 7 above)
Comment (jansenk) [reply openassessment/xblock/team_mixin.py:141]: Somewhat agreed, but a ticket for another day

File: openassessment/xblock/submission_mixin.py (hunk 4 above)
Comment (muselesscreator) [reply openassessment/xblock/submission_mixin.py:743]: +1
